utils/sample_qc.py

```diff
@@ -364,91 +364,109 @@ def infer_sex(
 
 
 def get_sex_expr(
-    chr_x_ploidy: hl.expr.NumericExpression,
-    chr_y_ploidy: hl.expr.NumericExpression,
-    f_stat: hl.expr.NumericExpression,
-    xx_x_ploidy_cutoffs: Tuple[float, float] = (1.4, 2.25),
-    xy_x_ploidy_cutoffs: Tuple[float, float] = (0.5, 1.4),
-    xx_y_ploidy_cutoff: float = 0.1,
-    xy_y_ploidy_cutoffs: Tuple[float, float] = (0.15, 1.2),
-    yy_y_ploidy_cutoff: float = 1.3,
-    xxx_x_ploidy_cutoff:float = 2.5,
-    f_stat_female_cutoff: float = -0.2,
-    f_stat_male_cutoff: float = 0.2
+    ht: hl.Table,
 ) -> hl.expr.StructExpression:
-    # TODO: Automate cutoffs
-    # TODO: Provide alternatives in case of missing annotations (e.g. no chr_y_ploidy)
     """
     Creates a struct with the following annotations:
-    - karyoptype (str)
-    - sex (str), which can be either 'male', 'female' or missing
+    - sex_karyoptype (str)
+    - sex (str), which can be either 'male', 'female', 'ambiguous_sex', or 'sex_aneuploidy'
     - is_female (missing for karyotypes that aren't either 'XX' or 'XY')
 
-    :param NumericExpression chr_x_ploidy: chrom X ploidy (or relative ploidy)
-    :param NumericExpression  chr_y_ploidy: chrom Y ploidy (or relative ploidy)
-    :param NumericExpression f_stat: chrom X F-stat
-    :param Tuple[float, float] xx_x_ploidy_cutoffs: Boundaries around the chom X ploidy for females
-    :param Tuple[float, float] xy_x_ploidy_cutoffs: Boundaries around the chom X ploidy for males
-    :param float xx_y_ploidy_cutoff: Max y ploidy for females
-    :param Tuple[float, float] xy_y_ploidy_cutoffs: Boundaries around the chom Y ploidy for males
-    :param float yy_y_ploidy_cutoff: Min chrom Y ploidy for YY
-    :param float xxx_x_ploidy_cutoff: Min chrom X ploidy for XXX
-    :param float f_stat_female_cutoff: Min f-stat for females
-    :param flat f_stat_male_cutoff: Max f-stat for males
-    :return: Struct expression with sex anotations
+    :param Table ht: Table with samples, their imputed sex, and their sex chromosome ploidies
     :rtype: StructExpression
     """
+    def get_ploidy_cutoffs(ht: hl.Table, sex: str) -> dict:
```

Comment:
Although this is clearly internal, I think having the number of SDs as args here could make sense

Reply:
Also, we may actually want to promote these as args to the main function, so that if these are too lax/stringent in a future callset they'd be easy to adjust

Reply:
great idea!